Add caching to NIP-05 validations

martindsq opened the pull request:
## Issues covered
#1215 

## Description

`URLSession` has a delete with a function that is invoked when a response will be cached so the developer can opt-out or modify the response that will be cached per request. However, that won't be called if the `URLSession` thinks it shouldn't be cached by reading the protocol rules. This is what happens in our cases, most NIP-05 servers aren't using the Cache-Control header so, per HTTP rules, the responses should not be cached.

The cleanest solution in our case is to use our custom cache instance and read/store the responses programmatically. `URLCache` is already good enough but we don't want to mess with the shared `URLCache` instance as we will want to wipe the contents of the cache, so, we make a new instance just for the URLSession that will be used for making NIP-05 validations.

## How to test
1. Navigate to Discover screen
2. Realize that the users there have their NIP-05 validated
3. Disconnect from the Internet
4. Navigate to the Profile of one of them
5. The NIP-05 also looks as validated there

joshuatbrown reviewed (commented):
This works, and the code looks fine, but when I follow the steps to reproduce the issue on `main`, I never see the strikethrough for the NIP-05. I thought that's how we showed unverified NIP-05 names.

martindsq commented:
@joshuatbrown I pushed a commit that addresses some feedback I got, you can see my relevant reply here https://planetary-app.slack.com/archives/CERN44F17/p1719342356129649?thread_ts=1719252929.642009&cid=CERN44F17

Aside from that, I still can reproduce the issue with the steps I wrote on `main`. Are you turning off both the wifi and cellular data? I did my tests in the simulator.

martindsq commented on Nos/Views/NIP05View.swift:
```diff
@@ -36,23 +36,20 @@ struct NIP05View: View {
                             username: nip05Identifier,
                             publicKey: publicKey
                         )
-                    } catch URLError.cancelled {
```
I'm reverting the logic here. Instead of showing the strikethrough in all errors and catch those in which we don't want to display the strikethrough, I'm now catching those we do want to display the strikethrough. The reason is because the list of URLError codes is very large (https://developer.apple.com/documentation/foundation/urlerror) and in most of them we don't want to display the strikethrough, also, false negatives are worse for the user experience than false negatives (its very bad to display a strikethrough for a user that has a valid nip-05).

So, from now on, if we detect another case for which we should display the strikethrough, we should add those in this try and catch. A added two that could be thrown if the user as a NIP-05 for a server that doesn't exist or for a server that is not accepting connections.

joshuatbrown reviewed (approved):
I'm able to reproduce the issue on `main` using a real device, and I verified that this PR fixes it. Everything looks great here!
